test/e2e/test_cluster_configs/flatcar/flatcar.json

```diff
@@ -0,0 +1,59 @@
+{
+	"apiModel": {
+		"apiVersion": "vlabs",
+		"properties": {
+			"orchestratorProfile": {
+				"orchestratorType": "Kubernetes",
+				"kubernetesConfig": {
+					"addons": [
+						{
+							"name": "cluster-autoscaler",
+							"enabled": true,
+							"pools": [
+								{
+									"name": "pool1",
+									"config": {
+										"min-nodes": "3",
+										"max-nodes": "9"
+									}
+								}
+							],
+							"config": {
+								"scan-interval": "1m",
+								"scale-down-delay-after-add": "1m0s",
+								"scale-down-unneeded-time": "2m0s"
+							}
+						}
+					]
+				}
+			},
+			"masterProfile": {
+				"count": 1,
+				"dnsPrefix": "",
+				"vmSize": "Standard_D2_v3"
+			},
+			"agentPoolProfiles": [
+				{
+					"name": "pool1",
+					"count": 3,
+					"vmSize": "Standard_D2_v3",
+					"distro": "flatcar"
+				}
+			],
+			"linuxProfile": {
+				"adminUsername": "azureuser",
+				"ssh": {
+					"publicKeys": [
+						{
+							"keyData": ""

```

Comment:
@jackfrancis Is the doc at https://github.com/Azure/aks-engine/blob/master/docs/topics/clusterdefinitions.md#notes-on-ssh-public-keys out of date? That doc suggests that at least 1 ssh pubkey is required for `linuxProfile`.

Reply:
Good question. The test runner that we ship as part of the AKS Engine project interprets these empty strings as "I'll create ssh keys for you", so that we don't have to publish public keys in public.

Something similar is happening for the empty string service principal values below.